test/Generator/ParameterGeneratorTest.php

```diff
@@ -591,6 +590,46 @@ public function testGetInternalClassDefaultParameterValue()
         self::assertSame('null', strtolower((string) $parameter->getDefaultValue()));
     }
 
+    /**
+     * @requires PHP >= 8.0
+     * @group laminas/laminas-code#53
+     *
+     * @dataProvider php80Methods
+     *
+     * @psalm-param class-string $className
+     * @psalm-param non-empty-string $method
+     * @psalm-param positive-int|0 $parameterIndex
+     * @psalm-param non-empty-string $expectedGeneratedSignature
+     */
+    public function testGeneratedSignatureForPhp80ParameterType(
+        string $className,
+        string $method,
+        int $parameterIndex,
+        string $expectedType,
+        string $expectedGeneratedSignature
+    ): void {
+        $parameter = ParameterGenerator::fromReflection(new ParameterReflection([$className, $method], $parameterIndex));
+
+        self::assertSame($expectedType, $parameter->getType());
+        self::assertSame($expectedGeneratedSignature, $parameter->generate());
+    }
+
+    /**
+     * @psalm-return non-empty-list<int, class-string, non-empty-string, positive-int|0, string, non-empty-string>
+     */
+    public function php80Methods(): array
+    {
+        return [
+            [Php80Types::class, 'mixedType', 0, 'mixed', 'mixed $parameter'],
+            [Php80Types::class, 'falseType', 0, Php80Types::class . '|false', '\\' . Php80Types::class . '|false $parameter'],
+            [Php80Types::class, 'unionNullableType', 0, 'bool', '?bool $parameter'],
+            [Php80Types::class, 'unionReverseNullableType', 0, 'bool', '?bool $parameter'],
+            [Php80Types::class, 'unionNullableTypeWithDefaultValue', 0, 'bool|string|null', 'bool|string|null $parameter = null'],
+            [Php80Types::class, 'unionType', 0, Php80Types::class . '|' . stdClass::class, '\\' . Php80Types::class . '|\\' . stdClass::class . ' $parameter'],
+            [Php80Types::class, 'selfAndBoolType', 0, Php80Types::class . '|bool', '\\' . Php80Types::class . '|bool $parameter'],
```

Comment:
Would be nice to have description keys for each of these; makes searching for specific cases on failure easier. The second string could very easily be re-purposed as a key for each of these.

Reply:
Similar to https://github.com/laminas/laminas-code/pull/57#discussion_r532898401, adding keys makes it more readable here, but harder to understand on failure.